base/random.jl

```diff
@@ -1224,6 +1227,14 @@ let Floats = Union{Float16,Float32,Float64}
             $randfun(                                dims::Integer...               ) = $randfun(GLOBAL_RNG, Float64, dims...)
         end
     end
+
+    for (T,SQRT_HALF) in ((Float64, 0.7071067811865475),
+                          (Float32, 0.70710677f0),
+                          (Float16, 0.70710677f0))
+        @inline function randn(rng::AbstractRNG, ::Type{Complex{T}})
+            Complex{T}(SQRT_HALF*randn(rng, T), SQRT_HALF*randn(rng, T))
+        end
+    end
```

Comment:
I wonder if it would be better to have:
```jl
@irrational SQRT_HALF 0.7071067811865475244008  sqrt(big(0.5))
randn{T}(rng::AbstractRNG, ::Type{Complex{T}}) = Complex{T}(SQRT_HALF*randn(rng, T), SQRT_HALF*randn(rng, T))
```
That way, it will work automatically for any real type `T`.  e.g. if we add BigFloat `randn` in the future, we will automatically get `Complex{BigFloat}`.

Also it is less code.